website/pages/how-to-guides/attack-surface-management-with-graph.md

```diff
@@ -0,0 +1,237 @@
+
+```sql
+MATCH 
+  (iamusers:aws_iam_users), 
+  (attachp:aws_iam_user_attached_policies)
+WHERE iamusers.arn = inlinep.user_arn
+CREATE (iamusers)-[r:has_attached_policy]->(attachp)
+RETURN type(r)
+```
+
+Next, we'll create relationships between IAM Groups and their inline policies.
+
+```sql
+MATCH 
+  (iamgroups:aws_iam_groups),
+  (groupinline:aws_iam_group_policies)
+WHERE iamgroups.arn = groupinline.group_arn
+CREATE (iamgroups)-[r:has_inline_policy]->(groupinline)
+RETURN type(r)
+```
+
+Lastly, we'll create relationships between IAM Groups and their attached managed policies.
+
+```sql
+MATCH 
+  (n:aws_iam_groups), 
+  (policies:aws_iam_policies) 
+UNWIND (keys(apoc.convert.fromJsonMap(n.policies))) as y 
+WITH y, policies, n
+WHERE y = policies.arn
+CREATE (n)-[r:has_attached_policy]->(policies)
+RETURN type(r)
+```
+
+After all these relationships have been created, we can run a `MATCH (n:aws_iam_users) return n` to return all IAM Users.  
+
+In the UI, feel free to play around with node labels, colors, and expansion of nodes and relationships. 
+
+In our sample environment, we have 3 IAM Users.  The following image shows the following and their relationships:
+* IAM User Access Keys in green.
+* IAM Managed Policies and Inline Policies in red.
+* IAM Users are in gray.
+* IAM Groups are in blue.
+
+We'll use the following query to show our IAM Users: `MATCH (n:aws_iam_users) return n`:
+
+![Sample Graph of IAM Users](/images/how-to-guides/attack-surface-management-with-graph/graph-users.png)
+
+#### Example 2: Data in RDS
+
+In this example, we will focus on Relational Databases (AWS RDS) and their data.  This will include network connectivity such as VPCs, Internet Gateways, and Security Groups.  We will also look at possible data access and encryption via KMS Keys, their KMS Key Policies, and KMS Key Grants.  In this example, we've already created the following resources in AWS: RDS Databases, KMS Keys, KMS Key Policies, KMS Key Grants, an AWS VPC, Security Groups, and an Internet Gateway.
+
+
+Let's first create relationships between RDS instances and their encryption from KMS Keys.
+
+```sql
+MATCH 
+  (rdsinstances:aws_rds_instances), 
+  (kmskeys:aws_kms_keys)
+WHERE rdsinstances.kms_key_id = kmskeys.arn
+CREATE (rdsinstances)-[r:is_encrypted_by_key]->(kmskeys)
+RETURN type(r)
+```
+
+Let's connect KMS Keys with all their Key Grants and the [access that the Key Grants may permit](https://www.cloudquery.io/blog/aws-kms-key-grants-deep-dive) to those KMS Keys and data.
+
+```sql
+MATCH 
+  (keygrants:aws_kms_key_grants),
+  (kmskeys:aws_kms_keys)
+WHERE keygrants.key_arn = kmskeys.arn
+CREATE (kmskeys)-[r:has_kms_key_grant]->(keygrants)
+RETURN type(r)
+```
+
+Let's now link Security Groups to the RDS instances.
+
+```sql
+MATCH 
+  (rds_is:aws_rds_instances), 
+  (sgs:aws_ec2_security_groups) 
+WHERE 
+  apoc.convert.fromJsonList(rds_is.vpc_security_groups)[0]['VpcSecurityGroupId'] 
```

Comment:
Could it be a problem that this seems to be selecting only the first entry in the list?